File: backend/set_role.py
import sys
import os

# Add the current directory to sys.path so it can find 'app' and 'models'
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from app import create_app
from models import User

def set_user_role(email, role):
    app = create_app()
    with app.app_context():
        user = User.objects(email=email).first()
        if not user:
            print(f"Error: User with email '{email}' not found.")
            return
        
        allowed_roles = ['student', 'mentor', 'admin']
        if role not in allowed_roles:
            print(f"Error: Invalid role '{role}'. Allowed roles are: {', '.join(allowed_roles)}")
            return
            
        user.role = role
        user.save()
        print(f"Success: {user.name}'s role has been updated to '{role}'.")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: python set_role.py <email> <role>")
    else:
        set_user_role(sys.argv[1].strip().lower(), sys.argv[2].strip().lower())

File: backend/src/scripts/setRole.js
/**
 * scripts/setRole.js
 * CLI script to update user role.
 * Usage: node src/scripts/setRole.js <email> <role>
 */

import dotenv from 'dotenv';
import connectDB from '../config/db.js';
import { User } from '../models/index.js';
import mongoose from 'mongoose';

dotenv.config();

async function setUserRole(email, role) {
  await connectDB();

  const user = await User.findOne({ email: email.toLowerCase() });
  if (!user) {
    console.error(`Error: User with email '${email}' not found.`);
    await mongoose.disconnect();
    return;
  }

  const allowedRoles = ['student', 'mentor', 'admin', 'reviewer', 'superadmin'];
  if (!allowedRoles.includes(role)) {
    console.error(`Error: Invalid role '${role}'. Allowed roles are: ${allowedRoles.join(', ')}`);
    await mongoose.disconnect();
    return;
  }

  user.role = role;
  await user.save();
  console.log(`Success: ${user.name}'s role has been updated to '${role}'.`);

  await mongoose.disconnect();
}

const args = process.argv.slice(2);
if (args.length < 2) {
  console.log('Usage: node src/scripts/setRole.js <email> <role>');
  process.exit(1);
}

setUserRole(args[0].trim(), args[1].trim()).catch((err) => {
  console.error('Error:', err);
  process.exit(1);
});
